aws-analytics-pinpoint/src/main/java/com/amplifyframework/analytics/pinpoint/AmazonPinpointAnalyticsPlugin.java

```diff
@@ -79,8 +93,71 @@ public void enable() {
      * {@inheritDoc}
      */
     @Override
-    public void identifyUser(@NonNull String userId, @NonNull AnalyticsProfile profile) {
-        throw new UnsupportedOperationException("This operation has not been implemented yet.");
+    public void identifyUser(@NonNull String userId, @NonNull AnalyticsUserProfile profile) {
+        EndpointProfile endpointProfile = targetingClient.currentEndpoint();
+        // Assign userId to the endpoint.
+        EndpointProfileUser user = new EndpointProfileUser();
+        user.setUserId(userId);
+        endpointProfile.setUser(user);
+        // Add user-specific data to the endpoint
+        addUserProfileToEndpoint(endpointProfile, profile);
+        // update endpoint
+        targetingClient.updateEndpointProfile();
+    }
+
+    /**
+     * Add user specific data from {@link AnalyticsUserProfile} to the endpoint profile.
+     * @param endpointProfile endpoint profile.
+     * @param userProfile user specific data to be added to the endpoint.
+     */
+    private void addUserProfileToEndpoint(EndpointProfile endpointProfile,
+                                          AnalyticsUserProfile userProfile) {
+        endpointProfile.addAttribute("email", Arrays.asList(userProfile.getEmail()));
+        endpointProfile.addAttribute("name", Arrays.asList(userProfile.getName()));
+        endpointProfile.addAttribute("plan", Arrays.asList(userProfile.getPlan()));
```

Comment:
The thing in the value position here is called a "singleton list," a list with one element. Java `Collections` utility provides a more efficient utility for constructing these things, `Collections.singletonList(ofOneItem)`.